sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_send_async.py

```diff
@@ -155,15 +156,38 @@ async def test_send_with_partition_key_async(connstr_receivers):
         await client.send_batch(await client.create_batch())
 
     found_partition_keys = {}
+    reconnect_receivers = []
     for index, partition in enumerate(receivers):
-        received = partition.receive_message_batch(timeout=5000)
-        for message in received:
+        retry_total = 0
+        while retry_total < 3:
```

Comment:
@yunhaoling - when using the consumer client, the retry is built in. However, since we're using the `receive_message_batch` method in `uamqp` (which is not actually used anywhere in the EH/SB code) and initializing our own `ReceiveClient`, we need to implement this retry logic. Similar logic exists in the `test_reconnect.py` file.

Reply:
@mikeharder - as per @yunhaoling, the biggest reason is that the ConsumerClient uses push-based receiving, which means that it keeps receiving forever unless the user triggers a close. The ReceiveClient is pull-based receiving, so we can just receive for a length of time or a certain number of events. This makes it easy to test the ProducerClient. Switching over to using the ConsumerClient in the tests would take some work.